File: Leetcode Daily/Find the Town Judge.java
import java.util.*;
class Solution {
    public int findJudge(int n, int[][] trust) {
        List<List<Integer>> map = new ArrayList<List<Integer>>(n);
        for(int i = 0; i<n; i++) map.add(new ArrayList<Integer>());
        for(int[] i : trust) map.get(i[0]-1).add(i[1]-1);
        int par = -1;
        for(int i = 0; i<n; i++){
            if (map.get(i).isEmpty()){
                par = i;
                break;
            }
        }
        for(int i = 0; i<n; i++){
            if (!map.get(i).contains(par) && i!=par) return -1;
        }
        return par+1;
    }
}   

File: Leetcode Daily/Find the Town Judge.rs
use std::collections::HashSet;

impl Solution {
    pub fn find_judge(n: i32, trust: Vec<Vec<i32>>) -> i32 {
        let mut trusters = HashSet::new();
        let mut trusted = vec![0; n as usize];
        for t in trust {
            trusters.insert(t[0]);
            trusted[t[1] as usize - 1] += 1;
        }
        for i in 0..n as usize {
            if trusted[i] == n - 1 && !trusters.contains(&(i as i32 + 1)) {
                return i as i32 + 1;
            }
        }
        -1
    }
}
